expression/builtin_encryption_vec.go

```diff
@@ -248,11 +250,85 @@ func (b *builtinSHA2Sig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinCompressSig) vectorized() bool {
-	return false
+	return true
+}
+
+var (
+	bytePool = sync.Pool{
+		New: func() interface{} {
+			return make([]byte, defaultChunkSize)
+		},
+	}
+)
+
+func allocByteSlice(n int) []byte {
+	if n > defaultChunkSize {
+		return make([]byte, n)
+	}
+	return bytePool.Get().([]byte)
+}
+
+func deallocateByteSlice(b []byte) {
+	if cap(b) <= defaultChunkSize {
+		bytePool.Put(b)
+	}
 }
 
+// evalString evals COMPRESS(str).
+// See https://dev.mysql.com/doc/refman/5.7/en/encryption-functions.html#function_compress
 func (b *builtinCompressSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+
+		str := buf.GetString(i)
+
+		// According to doc: Empty strings are stored as empty strings.
+		if len(str) == 0 {
+			result.AppendString("")
+		}
+
+		compressed, err := deflate([]byte(str))
```

Comment:
s/ `[]byte(str)`/ `hack.Slice(str)`

Reply:
sorry, i don't know what this mean. Could you give me more info

Reply:
change `[]byte(str)` to `hack.Slice(str)`

Reply:
thank you